internal/olm/operator/registry/operator_installer.go

```diff
@@ -122,49 +124,83 @@ func (o OperatorInstaller) waitForCatalogSource(ctx context.Context, cs *v1alpha
 	return nil
 }
 
-// createOperatorGroup creates an OperatorGroup using package name if an OperatorGroup does not exist.
-// If one exists in the desired namespace and it's target namespaces do not match the desired set,
-// createOperatorGroup will return an error.
-func (o OperatorInstaller) createOperatorGroup(ctx context.Context) error {
-	targetNamespaces := make([]string, len(o.InstallMode.TargetNamespaces), cap(o.InstallMode.TargetNamespaces))
-	copy(targetNamespaces, o.InstallMode.TargetNamespaces)
+func (o OperatorInstaller) ensureOperatorGroup(ctx context.Context) (*v1.OperatorGroup, error) {
 	// Check OperatorGroup existence, since we cannot create a second OperatorGroup in namespace.
 	og, ogFound, err := o.getOperatorGroup(ctx)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	// TODO: we may need to poll for status updates, since status.namespaces may not be updated immediately.
-	if ogFound {
-		// targetNamespaces will always be initialized, but the operator group's namespaces may not be
-		// (required for comparison).
-		if og.Status.Namespaces == nil {
-			og.Status.Namespaces = []string{}
-		}
-		// Simple check for OperatorGroup compatibility: if namespaces are not an exact match,
-		// the user must manage the resource themselves.
-		sort.Strings(og.Status.Namespaces)
-		sort.Strings(targetNamespaces)
-		if !reflect.DeepEqual(og.Status.Namespaces, targetNamespaces) {
-			msg := fmt.Sprintf("namespaces %+q do not match desired namespaces %+q", og.Status.Namespaces, targetNamespaces)
-			if og.GetName() == operator.SDKOperatorGroupName {
-				return fmt.Errorf("existing SDK-managed operator group's %s, "+
-					"please clean up existing operators `operator-sdk cleanup` before running package %q", msg, o.PackageName)
+
+	supported := o.SupportedInstallModes
+
+	if supported.Len() == 0 {
+		return nil, fmt.Errorf("operator %q is not installable: no supported install modes", o.StartingCSV)
```

Comment:
This error should be raised in validation before any in-cluster logic is run.

Reply:
Moved the logic early in the install.go